File: p274.cpp
//
// Created by iznauy on 18/04/2019.
//
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int hIndex(vector<int>& citations) {
        sort(citations.begin(), citations.end());
        reverse(citations.begin(), citations.end());
        for (int i = 0; i < citations.size(); i++) {
            if (citations[i] < i + 1)
                return i;
        }
        return citations.size();
    }
};

File: P274.java
import java.util.Arrays;

/**
 * Created on 18/04/2019.
 * Description:
 *
 * @author iznauy
 */
public class P274 {

    public int hIndex(int[] citations) {
        Arrays.sort(citations);
        for (int i = citations.length - 1; i >= 0; i--) {
            if (citations[i] < citations.length - i) {
                return citations.length - i + 1;
            }

        }
        return citations.length;
    }

}
